# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gnomesu/password-label.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Although we could have internationalized the file, specifing the only other
visible label comes handy as well.

  - Rene Rebe <rene@exactcode.de>

--- gnomesu-0.3.1/src/xsu.c.vanilla	2007-08-27 19:05:56.000000000 +0200
+++ gnomesu-0.3.1/src/xsu.c	2007-08-27 19:19:24.000000000 +0200
@@ -296,6 +296,7 @@
 	GtkWidget*	gtk_xsu_window;
 	gchar*		arg_command;
 	gchar*		arg_username=NULL;
+	gchar*		arg_message_password=NULL;
 	gint 	  x;
 	gboolean command_in=FALSE;
 	gboolean username_in=FALSE;
@@ -407,6 +408,20 @@
 				message_in=TRUE;
 			}
 		}
+
+		if (!strcmp (argv[x], "--message-password"))
+		{
+			if (argv[x+1] != NULL)
+			{       
+				int i;
+				arg_message_password = g_strdup(argv[x+1]);
+				for (i=0;i<strlen(arg_message_password);++i)
+					if (arg_message_password[i]=='^') arg_message_password[i]='\n';
+				for (i=0;i<strlen(arg_message_password);++i)
+					if (arg_message_password[i]=='~') arg_message_password[i]='\t';
+			}
+		}
+
 /* xsu 0.1.5 *
 	Option to load another icon
 
@@ -544,6 +559,11 @@
 	{
 		gtk_label_set_markup(GTK_LABEL(gtk_text_label),arg_message);
 	}
+
+	if (arg_message_password)
+	{
+		gtk_label_set_markup(GTK_LABEL(glade_xml_get_widget (xml, "gtk_password_label")),arg_message_password);
+	}
 	
 	if (command_in)
 	{
